plugins/in_kafka/in_kafka.c

```diff
@@ -246,6 +256,20 @@ static int in_kafka_init(struct flb_input_instance *ins,
         goto init_error;
     }
 
+    conf = flb_input_get_property("data_format", ins);
+    if (!conf) {
+        conf = "none";
+    }
+
+    if (!strcmp(conf, "none")) {
```

Comment:
This has to be a caseless comparison and comparison against zero should be explicit